frontend/src/pages/signUp/SignUp.tsx

```diff
@@ -20,20 +21,36 @@ const SignUp = () => {
 
   const { isDirty, isValid } = formState;
 
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [editingPass, setEditingPass] = useState<boolean>(false);
+
   const navigate = useNavigate();
 
   const onSubmit: SubmitHandler<FormData> = async (data) => {
     try {
       await createUserWithEmailAndPassword(auth, data.email, data.password);
+      setErrorMessage('');
       navigate('/login');
-    } catch (err) {
-      console.error(err);
+    } catch (e: any) {
+      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+        setErrorMessage(`Email address '${data.email}' is already in use `);
+      } else if (e.message) {
+        setErrorMessage(e.message);
+      } else {
+        setErrorMessage('Something went wrong. Please try again');
+      }
```

Comment:
This code is duplicated in LoginPage.tsx, can you extract the error handling elsewhere?